feat: Add membership-based resource limits

SharolRI opened the pull request:
- Link localities to memberships for user/connection limits
- Enforce user registration limits based on membership plans
- Implement water connections limits per membership
- Restrict assignable roles to supervisor and secretaria
- Add validation across UserController and WaterConnectionController
- Update database schema with locality-membership relationship

SharolRI committed:
feat: Add membership-based resource limits

- Link localities to memberships for user/connection limits
- Enforce user registration limits based on membership plans
- Implement water connections limits per membership
- Restrict assignable roles to supervisor and secretaria
- Add validation across UserController and WaterConnectionController
- Update database schema with locality-membership relationship

And-Y21 commented on database/seeders/LocalitiesTableSeeder.php:
```diff
@@ -11,24 +11,34 @@ class LocalitiesTableSeeder extends Seeder
                 'state' => 'Pennsylvania',
                 'zip_code' => '18503',
+                'membership_id' => $memberships->where('name', 'Basic Plan - 3 Months')->first()->id ?? $memberships->first()->id,
                 'expired' => true
```
Puedes poner solamente las id de las membresias

And-Y21 commented on database/seeders/LocalitiesTableSeeder.php:
```diff
@@ -51,11 +64,10 @@ public function run()
-            $query->whereNull('token')->orWhere('token', '');
-        })->get()->each(function ($locality) {
-            $locality->token = Token::generateTokenForLocality($locality->id);
-            $locality->save();
```
Esto no es necesario quitarlo, puedes agregar la membresia directamente ahi y validar el token

And-Y21 commented on resources/views/localities/tokenModal.blade.php:
```diff
@@ -20,16 +27,23 @@
+                                    {{ $membership->name }} - 
+                                    {{ $membership->term_months }} meses - 
+                                    {{ $membership->users_number }} usuarios - 
+                                    {{ $membership->water_connections_number }} conexiones
```
Son tomas de agua no conexiones

SharolRI commented on app/Http/Controllers/UserController.php:
```diff
@@ -23,26 +22,73 @@ public function index()
+            }
 
-        $user->roles()->sync($request->roles);
+            $currentUsersCount = \App\Models\User::where('locality_id', $request->locality_id)->count();
```
Esto me parece que es un error de github... porque en mi código aparece bien

SharolRI committed:
# Refactor: Optimize UserController and Localities Seeder

## Changes Made

### 1. UserController Refactor (pp/Http/Controllers/UserController.php)

#### Role Validation Optimization
- **Before:** Complex role validation using rray_diff with post-validation error handling
- **After:** Simplified role filtering using direct database queries with whereIn
- **Files Modified:**
  - pp/Http/Controllers/UserController.php - Methods: index(), store(), update(), edit()

**Specific Changes:**
- Removed redundant role validation logic in store() and update() methods
- Implemented proactive role filtering in index() and edit() methods:
  `php
  // Before:  = Role::all();
  // After:
   = Role::whereIn('name', ['Supervisor', 'Secretaria'])->get();
